### ISSUE TITLE
Feature Request: Add GitLab Statistics

### ISSUE THREAD

**Issue by TheCodersHangout** [pypa/warehouse#4301]
**What's the problem this feature will solve?**

Add **GitLab** Statistics

**Describe the solution you'd like**

Like GitHub Statistics, GitLab Statistics will be good if it is implemented, because gitlab has lot of pypi projects hosted!

**Comment by rixx** [comment]
This would probably only encompass gitlab.com at first, I think? Or do we have a way to figure out if a project is hosted on a GitLab instance?

**Comment by nanuxbe** [comment]
@nlhkabu I'm starting on this

**Comment by di** [comment]
@rixx Correct, let's just implement this for projects hosted on gitlab.com.

**Comment by nlhkabu** [comment]
Right now, we've noticed that GitHub statistics are broken :( They seem to appear once, then disappear when the page is refreshed.  We have some folks working on fixing this in the sprints, so we should perhaps not move onto this issue until that bug is fixed.

**Comment by di** [comment]
GitHub stats work by making an additional anonymous request from the users browser, and there is a limit to how many of these request can be made from a single IP.

So if a lot of people on a single network are looking at PyPI a lot, (e.g. at a Sprint) we might be hitting that limit.

IIRC you can see the request fail in the browser's console, and one way to work around this is to tether to a mobile device or join some other network.

**Comment by nlhkabu** [comment]
Yes, we just worked this out and are opening an issue for it. Not sure that it's a major problem, but worth discussing potential workarounds. 

In the meantime, we have a small team here working on GitLab stats :)
